tests/test_service.py

```diff
@@ -544,3 +544,41 @@ def test_deliver_order_item_with_delivered_status(self):
         new_item_id = new_order_json["order_items"][0]["item_id"]
         resp = self.app.put("/orders/{}/items/{}/deliver".format(new_order_id, new_item_id))
         self.assertEqual(resp.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_deliver_order(self):
+        """ deliver an order """
+        order_factory = _get_order_factory_with_items(2)
+        order_factory.order_items[0].status = "SHIPPED"
+        order_factory.order_items[1].status = "SHIPPED"
+
+        new_order_id = self._create_new_order(order_factory)["id"]
+
+        resp = self.app.put("/orders/{}/deliver".format(new_order_id))
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+
+    def test_deliver_order_not_found(self):
+        """ deliver an order when order does not exist"""
+        resp = self.app.put("/orders/{}/deliver".format(0))
+        self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_deliver_order_with_all_delivered_canceled(self):
```

Comment:
The name of this test case suggests that all items in the order have already been delivered or are cancelled. However, you set the statuses to placed and cancelled. We need to rethink our test cases here.

You might want to think of our test cases such as:
1. All order items are already "DELIVERED" -> resp 202
2. All items are "CANCELLED" -> resp 400
3. At least one item is "PLACED" -> resp 400
4. Some items are "SHIPPED" all remaining order items are "DELIVERED" -> resp 202